packages/@aws-cdk/aws-ecs/lib/base/task-definition.ts

```diff
@@ -293,6 +293,39 @@ export class TaskDefinition extends TaskDefinitionBase {
     return this._executionRole;
   }
 
+  /**
+   * Returns the host port that match the provided container name and container port.
+   *
+   * @internal
+   */
+  public _findHostPort(target: LoadBalancerTargetOptions): HostPort {
```

Comment:
The `HostPort` you are building here is used to create an `ec2.Port` above; do you need the intermediate step, or can you streamline the process and remove the extra interface?

If I'm reading things correctly, you could theoretically collapse the use of this function to `targetGroup.registerConnectable(self, self._findHostPort(options))` above, which addresses the relaxed ordering constraint I called out and removes some conditional logic to interpret this result.

Reply:
Agree!